### PR TITLE
Allow L2 interface to accept AviaryValues objects as well as filepaths

### PR DESCRIPTION
**Pull request by jkirk5**
### Summary

The Level 2 interface, specifically `load_inputs()`, now can accept an `AviaryValues` object as an alternative to creating one by reading an input deck (csv). Some option processing that is only required for GASP-based or 2DOF analysis have been modified to not always run regardless of which methods were selected. 

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Comment by johnjasa** [comment]
Thanks Jason! Quick questions.

Is the L2 interface accepting `AviaryValues` objects supposed to be user-facing? Should it be documented? Could you please include or modify a test that loads in an `AviaryValues` object instead of using a .csv file?

It looks like some multi-engine work got added on here -- is that on purpose?

**Comment by jkirk5** [comment]
> Is the L2 interface accepting `AviaryValues` objects supposed to be user-facing? Should it be documented? Could you please include or modify a test that loads in an `AviaryValues` object instead of using a .csv file?

Yeah, I can (and should) add documentation to show this!!

> It looks like some multi-engine work got added on here -- is that on purpose?
Nope, accidentally sent those to this PR by accident. I just reverted the fork to the correct commit

File: aviary/interface/methods_for_level2.py [hunk 1]
```diff
@@ -163,14 +163,23 @@ def load_inputs(self, input_filename, phase_info=None, engine_builder=None):
         """
         ## LOAD INPUT FILE ###
         self.engine_builder = engine_builder
-        self.aviary_inputs, self.initial_guesses = create_vehicle(input_filename)
 
-        aviary_inputs = self.aviary_inputs
+        # Create AviaryValues object from file (or process existing AviaryValues object
+        # with default values from metadata) and generate initial guesses
+        aviary_inputs, initial_guesses = create_vehicle(aviary_inputs)
+        # aviary_inputs = self.aviary_inputs
```
Comment (crecine) [inline aviary/interface/methods_for_level2.py:170]: remove?

File: aviary/utils/process_input_decks.py [hunk 2]
```diff
@@ -150,11 +146,27 @@ def update_options(aircraft_values: AviaryValues(), initial_guesses):
...
     tuple: Updated aircraft values and initial guesses.
     """
+    GASP_defaults = AviaryValues()
+    GASP_defaults.set_val('INGASP.JENGSZ', val=4)
+    GASP_defaults.set_val('test_mode', val=False)
+    GASP_defaults.set_val('use_surrogates', val=True)
+    GASP_defaults.set_val('mass_defect', val=10000, units='lbm')
+    GASP_defaults.set_val('problem_type', val=ProblemType.SIZING)
+    GASP_defaults.set_val(Aircraft.Electrical.HAS_HYBRID_SYSTEM, val=False)
+    GASP_defaults.set_val(Aircraft.Design.RESERVES, val=4998)
```
Comment (crecine) [inline aviary/utils/process_input_decks.py:156]: I think this will be unnecessary once #36 is merged

File: aviary/utils/process_input_decks.py [hunk 3]
```diff
@@ -150,11 +146,27 @@ def update_options(aircraft_values: AviaryValues(), initial_guesses):
...
+
+    # overwrite GASP_defaults with values from aircraft_values, then replace
+    # aircraft_values with this merged set
+    GASP_defaults.update(aircraft_values)
+    aircraft_values = GASP_defaults
+
+    for var in initial_guesses:
+        if var in aircraft_values:
+            initial_guesses[var], _ = aircraft_values.get_item(var)
```
Comment (crecine) [inline aviary/utils/process_input_decks.py:165]: Is this necessary? the initial guesses should all be distinct names that are separate from those in aircraft_values

**Review by johnjasa — APPROVED** [review]

**Review by crecine — APPROVED** [review]

**Pull request merged**